main.tf

```diff
@@ -0,0 +1,79 @@
+resource "aws_cloudtrail" "trail" {
+  name           = "trail${replace(title(var.project), " ", "")}${title(var.environment)}"
+  s3_bucket_name = "${var.s3_bucket_name}"
+
+  enable_logging                = true
+  include_global_service_events = true
+  is_multi_region_trail         = false
+  enable_log_file_validation    = false
+
+  tags {
+    Name        = "${var.project}"
+    Environment = "${var.environment}"
+  }
+
+  depends_on = ["aws_s3_bucket.trail"]
```

Comment:
This dependency is necessary since the bucket is not an implicit dependency of the trail. Interestingly, it doesn't seem to mess things up when you skip the automatic bucket creation (i.e. when the `count` of `aws_s3_bucket.trail` is `0`).

Reply:
I see. We couldn't count on `aws_s3_bucket.trail` in the event that it isn't created.